src/ranges.rs

```diff
@@ -131,6 +131,25 @@ impl RangeSet {
     }
 }
 
+impl PartialEq<Range<u64>> for RangeSet {
+    fn eq(&self, other: &Range<u64>) -> bool {
```

Comment:
Would this function return false when comparing two empty ranges for equivalence? Or is it not possible to create such a RangeSet?

Reply:
Hmmm, I should add a test for that.

Reply:
Oh but actually `RangeSet` can be empty, but `Range` can't as it's literally just a struct with two fields (start and end).

I should probably add a comment on what this whole thing does, but to clarify, this is comparison between `RangeSet` (which is a custom struct implemented in this file) and `Range` (which is provided by Rust standard library).

Reply:
well now you said it, it is clear from the method signature the equivalence check does that. I'm just less familiar with Rust. The comment helps though!